### ISSUE TITLE
show metabox on all post edit screens, but only edit for one post id

### ISSUE THREAD

**Issue by andrewatduckpin** [WebDevStudios/CMB2#740]
This is probably an unusual circumstance but...

I have posts with meta boxes on the edit screen in WP admin. One of those metaboxes is a textarea for "why choose us?"

I want the "why choose us?" metabox to be editable from the post editor (regardless of post id) but I want the information to pull from and save to just ONE post id.

So, if I edited the "why choose us?" paragraph on post 22, then opened the post editor for post 33, I would see the updated paragraph.

Basically, I would be fine with the information always saving to post 1 (regardless of what post 1 is), but it could be edited from any post editor screen.

Is there some sort of hook or hack I can use to make this work?

**Comment by tw2113** [comment]
Curious what you can do with the following filters.

For this first one, you'd need to save the data yourself to the meta field of choice, on the post ID of choice:
https://github.com/WebDevStudios/CMB2/blob/trunk/includes/CMB2_Field.php#L224-L260
Second filter would probably be best as you could run it ONLY on the field you intend, saves on having to conditionally ignore.

For this second one, you'd need to fetch the appropriate data from the appropriate post, and return it.
https://github.com/WebDevStudios/CMB2/blob/trunk/includes/CMB2_Field.php#L224-L260
Same thing regarding the second filter, it'd be specific to only the field you intend, and removes need for conditional checks.

Worth tinkering and trying out, at least.

**Comment by andrewatduckpin** [comment]
@tw2113 The two options appear to both be links to the first option. Could you update the second filter link to point to the relevant filter? Thanks for your suggestion

**Comment by tw2113** [comment]
Oops, you are correct. The 2 I originally linked are for the 2nd spot with regards to returned value

Correct link for the first one, regarding saving: 
https://github.com/WebDevStudios/CMB2/blob/trunk/includes/CMB2_Field.php#L290-L324

**Comment by andrewatduckpin** [comment]
@tw2113 thanks for the help

In case anyone visits looking for same filter solution... These two filters together will
1) Save metadata for the specified fields (in the $fields var) to post ID 1 (you can change this if you want) regardless of what the current post ID is.
2) Show metadata from post ID 1 in the specified fields (in the $fields var), regardless of what the current post ID is.

`function override_meta_save( $empty, $a, $args, $object ) {
	$fields = array("field_one", "field_two", "field_three");
	if(in_array($a['field_id'], $fields)){
  		update_post_meta(1, $a['field_id'], $a['value']);
	}
	return true;
}
add_filter( 'cmb2_override_meta_save', 'override_meta_save', 10, 4 );

function override_meta_show( $default, $objectid, $a, $object ) {
	$fields = array("field_one", "field_two", "field_three");
	if(in_array($a['field_id'], $fields)){
  		return get_post_meta( 1, $a['field_id'], true );
	}else{
  		return $default;
	}
}
add_filter( 'cmb2_override_meta_value', 'override_meta_show', 10, 4 );`

**Comment by tw2113** [comment]
Awesome, sounds like you got a version to work for this.
